[18.0][IMP] stock_release_channel: collect pickings

jbaudoux opened the pull request:
Allow to stop collecting pickings in a release channel. Non releasable deliveries are removed from the release channel and a job is spawn to try to reassign them to another channel.

At some point in time (typically when the cutoff is reached), you want to stop having new deliveries to release. Up to now, the only way was to sleep the release channel. However, you cannot do that when there is still released work to process. You need to process the "To Do" pickings & deliveries.
During this time, backorders could become available (thanks to goods reception) and generate more work to release. 
Also, if there is another channel in the future that allows to process the delivery, you may prefer to forward the delivery to that channel. In the current channel, you may not have the time to process the released work.

<img width="766" height="306" alt="image" src="https://github.com/user-attachments/assets/6836b134-8b3b-4c4b-a89b-628732c2b26a" />

New actions:
* Stop Collecting
* Restart Collecting

cc @sebalix @Highcooley @rousseldenis @lmignon @mt-software-de @phschmidt

jbaudoux committed:
[IMP] stock_release_channel: collect pickings

Clearly split the fact that a channel collects pickings from the state

jbaudoux committed:
[IMP] stock_release_channel_cutoff: collect pickings

Rely on new boolean for the warning

jbaudoux committed:
[IMP] stock_release_channel_partner_address: collect pickings

mt-software-de commented on stock_release_channel/models/stock_release_channel.py:
```diff
@@ -202,13 +202,16 @@ class StockReleaseChannel(models.Model):
     last_done_picking_name = fields.Char(compute="_compute_last_done_picking")
     last_done_picking_date_done = fields.Datetime(compute="_compute_last_done_picking")
+    collect_pickings = fields.Boolean(
+        compute="_compute_collect_pickings",
```
Isn't there are `default=True` missing, or a migration script to not break the current behavior?
Since `_compute_collect_pickings` only setting `False`

jbaudoux replied:
Yes I forgot the migration script that I wanted to do...

jbaudoux replied:
migration script added

mt-software-de reviewed (changes requested):
Thx. This look very helpful. 
Just a minor remark.

jbaudoux committed:
[IMP] stock_release_channel: recompute channel

Move recomputation option from company to channel as this recomputation
depends if lifecycle is used or not on the channel

jbaudoux commented:
@mt-software-de Can you update your review?

jbaudoux commented:
I added a commit moving the recompute of channel from company to channel. When you manage channels with lifecycle, you don't want this recomputation to occur. If the channel stops collecting, you cannot recompute after release otherwise the delivery will be kicked out of the channel.
This recomputation was there in very first version of the module before lifecycle was introduced. The channel should be properly set before releasing and this is already the case with carrier alternative or with this fiw https://github.com/OCA/stock-logistics-release-channel/pull/46

rousseldenis reviewed (approved):
Seems great !

mt-software-de commented on stock_release_channel/models/stock_release_channel.py:
```diff
@@ -51,6 +51,12 @@ class StockReleaseChannel(models.Model):
 
     name = fields.Char(required=True)
     release_forbidden = fields.Boolean(string="Forbid to release this channel")
+    recompute_channel_on_pickings_at_release = fields.Boolean(
```
Before your change this was by default True, since it was only configured on the company. 
Is it fine to change the behavior now?

jbaudoux replied:
The pre-migration script ensure it stays True for existing installation. It better be False by default as it's not really compatible with the lifecycle

mt-software-de reviewed (commented):
Just a minor question.

jbaudoux commented:
manually bumping version of other modified modules

jbaudoux commented:
/ocabot merge nobump
